### PR TITLE
Fix particle property vector output

### PR DESCRIPTION
**Pull request by gassmoeller**
The output of vector properties for particles had some quirks that originated in the assumption that vectors in vtk or hdf5 can have a different number of components than the number of spatial dimensions. This is not supported but sometimes worked in Paraview and Visit (e.g. if you have 3 components in a 2d model, they simply assume it is a 3d model), but often lead to errors in one or the other program. This change resolves this in the following way:
- If a particle property has `dim` components, it is interpreted and written as a vector with 3 components (at least the vtk formats only support 3D vectors), with the z component being zero.
- If a particle property has any other number of components it is written as `n_components` scalar fields with the names `name_componentindex`

Also, while updating the hdf5 output, I noticed that we did not make the name change from `particle` to `particles` in the output file names that we did for the vtk output. This PR resolves that issue.

### CODE REVIEW

**Comment by bangerth** [comment]
A bunch of tests seem to fail. Can you check?

**Comment by bangerth** [comment]
The code changes look reasonable to me, though I'm not going to claim that I fully understand them. We really need to get particle output into `DataOutBase` so we can, for example, benefit from data compression in VTU format.

OK to merge if you figure out what is going on with the tests.

**Comment by gassmoeller** [comment]
I updated the test results that this PR affected, but I have no clue what is happening with the clang tests. They show timeouts of melt tests, but this PR did not change anything there (they do not use particles). I would guess this is a tester issue.

File: tests/particle_output_hdf5/particles.xdmf [hunk 1]
```diff
@@ -0,0 +1,72 @@
+----------------
+##10      <==             particles/particle-00000.h5:/nodes
+##10      ==>             particles/particles-00000.h5:/nodes
+
+##10      #:1   <== particles/particle-00000.h5
+##10      #:1   ==> particles/particles-00000.h5
```
Comment (gassmoeller) [inline tests/particle_output_hdf5/particles.xdmf]: I am not sure where you saw these lines. Maybe this was a bug on github?
I have rebased one more time, but if the clang tests keep timing out I would say lets just merge nevertheless.

Comment (bangerth) [inline tests/particle_output_hdf5/particles.xdmf]: Yes, good question. Who knows. Let's just ignore this.

**Comment by bangerth** [comment]
OK to merge if the aspect-8.4.1b test is through.

**Comment by tjhei** [comment]
> but I have no clue what is happening with the clang tests.

Yeah, I think I finally understand what is going on and this problem is hopefully fixed now. You still have one failing test here, though.

**Comment by bangerth** [comment]
The failing test is the one fixed in the meantime in #1236. The base of @gassmoeller 's branch is simply before the merge point for #1236, so the failing test will not show up after the merge.

**Pull request merged**